src/pages/postgraphile/usage-schema.md

```diff
@@ -238,3 +238,288 @@ look at the `postgraphile-core` and `graphile-build-pg` modules.
 
 [graphql-js]: https://www.npmjs.com/package/graphql
 [`pg-pool`]: https://www.npmjs.com/package/pg-pool
+
+# Server side TypeScript support
+
+PostGraphile is taking care of the GraphQL API which different clients can use.
+But it is not only possible to use the API from external clients. It is also
+possible to use the GraphQL API from within its own backend.
+
+High-level overview:
+
+- use the graphql-code-generator to create TypeScript types for our GraphQL
+  schema
+- use the generated types to query/mutate your data
+- optional: use a Visual Studio Code extension to get IntelliSense
+
+## TypeScript Code generation
+
+We use the [{GraphQL} code generator](https://graphql-code-generator.com/) tools
+to create the TypeScript types in our backend code.
+
+The following `npm` packages will create typescript types:
+
+- `@graphql-codegen/cli` - the CLI tool to create the types
+- `@graphql-codegen/typescript` - create the typescript types. This is the main
+  package that we need
+- `@graphql-codegen/typescript-operations` - this input types for queries
+- optional: `@graphql-codegen/typescript-urql` - this produces typescript
+  queries from any custom `*.graphql` query/mutation files that we define.
+
+Steps to get the backend typing support:
+
+1. Start the development like described in the above section. Follow all the
+   steps to create your first GraphQL API with all needed schemas, tables,
+   roles, etc.
+2. Configure in the PostGraphile setup to export the GraphQL schema:  
+   `exportGqlSchemaPath: './src/generated/schema.graphql'`
+3. Start the project to let PostGraphile create the initial `schema.graphql`
+   file. You can exclude this file from source control if you want to. But it
+   might be nice to see differences in the schema during check-ins.
```

Comment:
```suggestion
3. Start the project to let PostGraphile create the initial `schema.graphql`
   file. Since it's generated you can exclude this file from source control if
   you want to, but it is handy to see differences in the schema during check-ins
   and is useful for other tooling such as `eslint-plugin-graphql`.
```